prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -58,10 +78,181 @@ class Hl7Serializer(val metadata: Metadata) {
         outputStream.write(createFooters(report).toByteArray())
     }
 
+    /*
+     * Read in a file
+     */
+    fun convertBatchMessagesToMap(message: String, schema: Schema): Hl7Mapping {
+        val mappedRows: MutableMap<String, MutableList<String>> = mutableMapOf()
+        val errors = mutableListOf<String>()
+        val warnings = mutableListOf<String>()
+        val rowResults = mutableListOf<RowResult>()
+        val reg = "(\r|\n)".toRegex()
+        val cleanedMessage = reg.replace(message, "\r")
+        val messageLines = cleanedMessage.split("\r")
+        val nextMessage = StringBuilder()
+
+        fun deconstructStringMessage() {
+            println("Examining:\n$nextMessage")
+            val parsedMessage = convertMessageToMap(nextMessage.toString(), schema)
+            errors.addAll(parsedMessage.errors)
+            warnings.addAll(parsedMessage.warnings)
+            rowResults.add(parsedMessage)
+            nextMessage.clear()
+            parsedMessage.row.forEach { (k, v) ->
+                if (!mappedRows.containsKey(k))
+                    mappedRows[k] = mutableListOf()
+
+                mappedRows[k]?.addAll(v)
+            }
+        }
+
+        messageLines.forEach {
+            if (it.startsWith("FHS"))
+                return@forEach
+            if (it.startsWith("BHS"))
+                return@forEach
+            if (it.startsWith("BTS"))
+                return@forEach
+            if (it.startsWith("FTS"))
+                return@forEach
```

Comment:
All of these segments are not supported by HAPI, so I am just throwing them away for now. Maybe in the future we can do something with them for validation. For now, however, let's just get rid of them.